elasticapm/instrumentation/packages/graphql.py

```diff
@@ -96,8 +96,12 @@ class GraphQLBackendInstrumentation(AbstractInstrumentedModule):
     ]
 
     def get_graphql_tx_name(self, graphql_doc):
-        op = graphql_doc.definitions[0].operation
-        fields = graphql_doc.definitions[0].selection_set.selections
+        op_def = [
```

Comment:
According to the [spec](https://spec.graphql.org/June2018/#sec-Language.Document), it's possible that documents don't have any `OperationDefinition`s 

> Documents are only executable by a GraphQL service if they contain an OperationDefinition and otherwise only contain ExecutableDefinition. However documents which do not contain OperationDefinition or do contain TypeSystemDefinition or TypeSystemExtension may still be parsed and validated to allow client tools to represent many GraphQL uses which may appear across many individual files.

I'm still not super familiar with GraphQL, but can we assume that `get_graphql_tx_name` is only called for documents that are actually executed? Otherwise we'd need to catch the `IndexError` here.